src/tallies/tally.cpp

```diff
@@ -381,6 +393,10 @@ void Tally::add_filter(Filter* filter)
     energyout_filter_ = filters_.size();
   } else if (dynamic_cast<const DelayedGroupFilter*>(filter)) {
     delayedgroup_filter_ = filters_.size();
+  } else if (dynamic_cast<const CellFilter*>(filter)) {
+    cell_filter_ = filters_.size();
+  } else if (dynamic_cast<const EnergyFilter*>(filter)) {
```

Comment:
Alright, I have a better solution here: instead of using `dynamic_cast`, you can use the `type()` method on filters, which is virtual. It will look like:
```C++
} else if (filter->type() == FilterType::ENERGY) {
```

Reply:
yes, this is more clear

Reply:
Yes, so I did the change in the lines already in the lines 447-454. I forgot that I also need to do it here. I changed now the check for the `CellFilter.` Actually I am not so happy with it, cause now we are using the `dynamic_cast` method and the `type()` method beside of each other...

We don't need to check the energy filter here, since earlier in this if condition it is checked if this is an `EnergyoutFilter`

Reply:
Yes, I would recommend we change all of them here to not use `dynamic_cast` and instead rely on `filt->type()` checks.